### PR TITLE
GH-3105: Spring integration stream SFTP: pattern filter not wrking

### PR DESCRIPTION
**Pull request by lukasGemela**
https://github.com/spring-projects/spring-integration/issues/3105

### CODE REVIEW

File: spring-integration-file/src/main/java/org/springframework/integration/file/remote/AbstractRemoteFileStreamingMessageSource.java [hunk 1]
```diff
@@ -190,9 +190,10 @@ protected Object doReceive() {
 
 					if (this.toBeReceived.size() > 0) { // don't re-fetch already filtered files
 						file = poll();
+						continue;
 					}
 					else {
-						file = null;
+						break;
```
Comment (artembilan) [inline spring-integration-file/src/main/java/org/springframework/integration/file/remote/AbstractRemoteFileStreamingMessageSource.java]: Grate!

I think we don't need an `if (file != null)` any more after this.

**Review by artembilan — CHANGES_REQUESTED** [review]
Please, add your name to the `@author` list.
And also it would be great to have some test to cover this functionality.

I guess if we would have had such a test before we wouldn't have this bug at all 😢 

Thank you for your contribution anyway!

File: spring-integration-file/src/main/java/org/springframework/integration/file/remote/AbstractRemoteFileStreamingMessageSource.java (hunk 1 above)
Comment (lukasGemela) [reply spring-integration-file/src/main/java/org/springframework/integration/file/remote/AbstractRemoteFileStreamingMessageSource.java]: Sure thing will do tomorrow

Comment (lukasGemela) [reply spring-integration-file/src/main/java/org/springframework/integration/file/remote/AbstractRemoteFileStreamingMessageSource.java]: @artembilan test added

**Review by artembilan — APPROVED** [review]
Grate!

Pulling locally for final review and merge after code clean up.

👍

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration/commit/1820c05c26ad625d449d49b867a72f8d1cf97571.

@lukasGemela ,

Thank you for contribution; looking forward for more!

**Pull request closed without merging**